scripts/quests/crystalWar/The_Dawn_of_Delectability.lua

```diff
@@ -0,0 +1,200 @@
+-- DRAGON_FRUIT    - !additem 5662
+--
+-- Ranpi-Monpi (S) - !pos -115 -3 43 94
+-- Ranpi-Monpi     - !pos -116 -3 52 238
+-----------------------------------
+require('scripts/globals/interaction/quest')
+require('scripts/globals/npc_util')
+require('scripts/globals/keyitems')
+require('scripts/globals/quests')
+require('scripts/globals/items')
+require('scripts/globals/zone')
+-----------------------------------
+local watersID = require("scripts/zones/Windurst_Waters_[S]/IDs")
+-----------------------------------
+
+local quest = Quest:new(xi.quest.log_id.CRYSTAL_WAR, xi.quest.id.crystalWar.THE_DAWN_OF_DELECTABILITY)
+
+quest.reward =
+{
+    item = xi.items.TRAINEE_KNIFE,
+}
+
+quest.sections =
+{
+    -- Visit Ranpi-Monpi (S) in Windurst Waters (S) (E-9, north side) to start the quest.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_AVAILABLE
+        end,
+
+        [xi.zone.WINDURST_WATERS_S] =
+        {
+            ['Ranpi-Monpi'] = quest:progressEvent(117),
+
+            onEventFinish =
+            {
+                [117] = function(player, csid, option, npc)
+                    quest:begin(player)
+                    quest:setVar(player, 'Prog', 1)
+                end,
+            },
+        },
+    },
+
+    -- Speak with Ranpi in the present and he'll request ingredients
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 1
+        end,
+
+        [xi.zone.WINDURST_WATERS_S] =
+        {
+            ['Ranpi-Monpi'] = quest:progressEvent(119), -- Optional dialogue
+        },
+
+        [xi.zone.WINDURST_WATERS] =
+        {
+            ['Ranpi-Monpi'] = quest:progressEvent(978),
+
+            onEventFinish =
+            {
+                [978] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 2)
+                end,
+            },
+        },
+    },
+
+    -- Trade Ranpi the ingredients
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 2
+        end,
+
+        [xi.zone.WINDURST_WATERS] =
+        {
+            ['Ranpi-Monpi'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHas(trade, {xi.items.PEPPERONI, xi.items.WALNUT,
+                        xi.items.DRAGON_FRUIT, xi.items.BASTORE_SWEEPER}) then
+                            return quest:progressEvent(983)
+                    end
+                end,
+
+                onTrigger = function(player, npc)
+                    return quest:progressEvent(979) -- reminder dialogue
+                end,
+            },
+
+            onEventFinish =
+            {
+                [983] = function(player, csid, option, npc)
+                    player:confirmTrade()
+                    quest:setVar(player, 'Prog', 3)
+                    quest:setVar(player, "DelectabilityDay", vanaDay())
```

Comment:
If you're looking for a unique day, there's a binding for this in luautils as well, `VanadielUniqueDay`